### PR TITLE
JAMES-2813 Add cancel support in DistributedTaskManager

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[ce9ff4f270d5c58849cd50759a3b8fc4c1259585] [INFO] There is 1 error reported by Checkstyle 8.18 with checkstyle.xml ruleset.
[ce9ff4f270d5c58849cd50759a3b8fc4c1259585] [ERROR] src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java:[29,8] (imports) UnusedImports: Unused import - org.apache.commons.lang3.NotImplementedException.
```

File: server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java [hunk 1]
```diff
@@ -42,24 +44,34 @@
...
+import reactor.rabbitmq.RabbitFlux;
+import reactor.rabbitmq.Receiver;
 import reactor.rabbitmq.ReceiverOptions;
 import reactor.rabbitmq.Sender;
 
 public class RabbitMQWorkQueue implements WorkQueue, Startable {
     private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQWorkQueue.class);
 
-    static final Integer MAX_CHANNELS_NUMBER = 1;
+    static final Integer MAX_CHANNELS_NUMBER = 5;
```
Comment (rouazana) [inline server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java]: why 5?

**Comment by rouazana** [comment]
test this please (no test results found)

File: server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java (hunk 1 above)
Comment (blackheaven) [reply server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java]: We need 2 for the listener plus 1-2 for the senders, I can set it to 3 if you want

Comment (rouazana) [reply server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java]: I'm more interested by the why than the value in fact :). I don't know at all which value is better, 3 or 5, so I let you choice the better. But can you add a comment explaining your choice?

File: server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java [hunk 2]
```diff
@@ -57,13 +56,13 @@
...
 import reactor.rabbitmq.QueueSpecification;
 import reactor.rabbitmq.RabbitFlux;
-import reactor.rabbitmq.Receiver;
 import reactor.rabbitmq.ReceiverOptions;
 import reactor.rabbitmq.Sender;
 
 public class RabbitMQWorkQueue implements WorkQueue, Startable {
     private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQWorkQueue.class);
 
+    // Need at least one by receivers but a shared one for senders
```
Comment (rouazana) [inline server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java]: s/but/plus/ ?

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**